openshift/machine-config-operator
# MCO-1097: Automate mechanism to gather the Simple Content Access Certificate/secret

**- What I did**
This creates a new function called in `syncMachineOSBuilder` loop which does:
- When any pool is opted into layering, a clone of the `etc-pki-entitlement` secret(in the `openshift-config-managed` namespace) will made in the MCO namespace with the name `etc-pki-entitlement-$(POOL_NAME)`.
- When the pool is opted out of layering, this cloned secret in the MCO namespace is deleted.
 
The build pod will mount the appropriate cloned secret for the pool it is building for.

**- How to verify it**
- Opt a pool into layering. Assuming the cluster has the RHEL entitlement secret, it should create a clone of it and do a successful build with this secret. 
- Opt the pool out of layering(delete associated MachineOSConfig), this should delete the cloned secret.
